diff --git a/node_modules/@projectstorm/react-canvas-core/dist/core-actions/ActionEventBus.js b/node_modules/@projectstorm/react-canvas-core/dist/core-actions/ActionEventBus.js
index 3bc5cb0..2064095 100644
--- a/node_modules/@projectstorm/react-canvas-core/dist/core-actions/ActionEventBus.js
+++ b/node_modules/@projectstorm/react-canvas-core/dist/core-actions/ActionEventBus.js
@@ -44,12 +44,12 @@ class ActionEventBus {
         }
         else if (event.type === 'keydown') {
             // store the recorded key
-            this.keys[event.key.toLowerCase()] = true;
+            this.keys[event.key ? event.key.toLowerCase() : 'Unidentified'] = true;
             return this.getActionsForType(Action_1.InputType.KEY_DOWN);
         }
         else if (event.type === 'keyup') {
             // delete the recorded key
-            delete this.keys[event.key.toLowerCase()];
+            delete this.keys[event.key ? event.key.toLowerCase() : 'Unidentified'];
             return this.getActionsForType(Action_1.InputType.KEY_UP);
         }
         else if (event.type === 'mousemove') {
diff --git a/node_modules/@projectstorm/react-canvas-core/dist/states/SelectionBoxState.js b/node_modules/@projectstorm/react-canvas-core/dist/states/SelectionBoxState.js
index 8fb8c5e..a923d7e 100644
--- a/node_modules/@projectstorm/react-canvas-core/dist/states/SelectionBoxState.js
+++ b/node_modules/@projectstorm/react-canvas-core/dist/states/SelectionBoxState.js
@@ -27,7 +27,7 @@ class SelectionBoxState extends AbstractDisplacementState_1.AbstractDisplacement
         }
         else if (event.event instanceof TouchEvent) {
             const touch = event.event.touches[0];
-            rel = this.engine.getRelativePoint(touch.clientX, touch.clientY);
+            rel = this.engine.getRelativePoint(touch.clientX, touch.clientY) || { x: 0, y: 0 };
         }
         return {
             left: rel.x > this.initialXRelative ? this.initialXRelative : rel.x,
